pkg/apiserver/auth/config_key.go

```diff
@@ -91,10 +89,20 @@ func (a *configKeyAuthenticator) IsValid(ginCtx *gin.Context) (bool, error) {
 	return true, nil
 }
 
-func (a *configKeyAuthenticator) getApiKeyFromSource(ginCtx *gin.Context) string {
-	if a.source == SourceQueryParams {
-		return ginCtx.Query(ByApiKey)
+func QueryProvider(queryName string) ApiKeyProvider {
```

Comment:
Can we find a better naming than `QueryProvider`? I'm not sure myself what speaks best for what it does, but this is too generic i think.